### PR TITLE
e-icons CSS: Tweak - e-icons classes are not needed in the frontend, only the font-face is needed - ED-950,ED-1190,ED-960

### PR DESCRIPTION
**Pull request by rotemee**
The e-icons classes are not needed in the frontend, they are only needed in the Admin, Editor, and the Frontend Preview Mode.
Therefore, we've separated the font-face call from the e-icons classes which will load the e-icons font files only when needed on the page. By adding the e-icons classes file to the assets loader, the classes will automatically be available in the frontend preview mode and will be loaded in the Admin and Editor by the App file.

## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: assets/dev/js/frontend/utils/swiper.js [hunk 1]
```diff
@@ -7,17 +7,17 @@ export default class Swiper {
 			this.config = this.adjustConfig( config );
 		}
 
-		return new Promise( ( resolve ) => {
-			if ( ! elementorFrontendConfig.environmentMode.isOptimizedJS ) {
-				return resolve( this.createSwiperInstance( container, this.config ) );
+		return new Promise( ( res ) => {
```
Comment (kobizz) [inline assets/dev/js/frontend/utils/swiper.js:10]: Change it back to `resolve`

File: assets/lib/eicons/css/elementor-icons.css [hunk 2]
```diff
@@ -1,10 +1,3 @@
-@font-face {
```
Comment (kobizz) [inline assets/lib/eicons/css/elementor-icons.css:1]: Return it back

File: assets/lib/eicons/css/elementor-icons.min.css [hunk 3]
```diff
@@ -1,2 +1,2 @@
...
...nt:'\e963'}.eicon-lightbox-expand:before{content:'\e964'}.eicon-error-404:before{content:'\e965'}.eicon-theme-style:before{content:'\e966'}.eicon-search-results:before{content:'\e967'}.eicon-single-post:before{content:'\e968'}.eicon-site-identity:before{content:'\e969'}.eicon-theme-builder:before{content:'\e96a'}.eicon-download-bold:before{content:'\e96b'}.eicon-share-arrow:before{content:'\e96c'}.eicon-global-settings:before{content:'\e96d'}.eicon-user-preferences:before{content:'\e96e'}.eicon-lock:before{content:'\e96f'}.eicon-export-kit:before{content:'\e970'}.eicon-import-kit:before{content:'\e971'}.eicon-lottie:before{content:'\e972'}.eicon-products-archive:before{content:'\e973'}.eicon-single-product:before{content:'\e974'}.eicon-disable-trash-o:before{content:'\e975'}.eicon-single-page:before{content:'\e976'}.eicon-wordpress-light:before{content:'\e977'}.eicon-cogs-check:before{content:'\e978'}.eicon-custom-css:before{content:'\e979'}.eicon-global-colors:before{content:'\e97a'}.eicon-globe:before{content:'\e97b'}.eicon-typography-1:before{content:'\e97c'}.eicon-background:before{content:'\e97d'}.eicon-eye:before{content:'\e8ac'}.eicon-elementor-square:before{content:'\e813'}
```
Comment (kobizz) [inline assets/lib/eicons/css/elementor-icons.min.css:2]: Same as above

File: core/utils/assets-loader.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+	private $assets;
+
+	public function get_name() {
+		return 'assets-loader';
+	}
+
+	private function init_assets() {
+		$this->assets = [
+			'styles' => [
+				'e-animations' => [
```
Comment (kobizz) [inline core/utils/assets-loader.php:28]: Tty to convert each asset to a class

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/js/frontend/utils/swiper.js (hunk 1 above)
Comment (rotemee) [reply assets/dev/js/frontend/utils/swiper.js]: Fixed.

Comment (rotemee) [reply assets/dev/js/frontend/utils/swiper.js]: Fixed.

File: assets/lib/eicons/css/elementor-icons.css (hunk 2 above)
Comment (rotemee) [reply assets/lib/eicons/css/elementor-icons.css:1]: Fixed.

File: assets/lib/eicons/css/elementor-icons.min.css (hunk 3 above)
Comment (rotemee) [reply assets/lib/eicons/css/elementor-icons.min.css:2]: Fixed.

File: core/utils/assets-loader.php (hunk 4 above)
Comment (rotemee) [reply core/utils/assets-loader.php:28]: As agreed, will be done in a different scope.

File: includes/base/controls-stack.php [hunk 5]
```diff
@@ -1081,6 +1081,34 @@ public function get_active_settings( $settings = null, $controls = null ) {
...
+				// Enabling assets loading from the control object.
+				$control_conditional_assets = $control_obj::get_assets( $setting );
+
+				if ( $control_conditional_assets ) {
+					foreach ( $control_conditional_assets as $assets_type => $dependencies ) {
+						foreach ( $dependencies as $dependency ) {
+							Plugin::$instance->assets_loader->enable_asset( $assets_type, $dependency );
+						}
+					}
+				}
```
Comment (matipojo) [inline includes/base/controls-stack.php:1111]: Move to the saving process, and save as a meta with the dependent assets.

The meta will be saved to `_elementor_assets`

Add an upgrade script to delete the old `_elementor_css` meta.

The `Plugin::$instance->files_manager->clear_cache()` will delete both `_elementor_assets` & `_elementor_css`. until v3.6.

**Review by matipojo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**